mla/src/crypto/aesgcm.rs

```diff
@@ -58,7 +61,7 @@ impl AesGcm256 {
         ghash.update_padded(associated_data);
 
         // Prepare the cipher for further operations
-        let mut cipher = aes_ctr::Aes256Ctr::new_var(key, &counter_block)?;
+        let mut cipher = Aes256Ctr::new(key.into(), &counter_block.into());
```

Comment:
Here, the `key` length is no more checked, and we use the conversion to `GenericArray< , U32>`.

Instead of removing the `InvalidCipherInit`, we could add an explicit check for `key` length and `nonce` length, and throw the error if it does not match.

It's likely better to add it in this PR, to avoid the remove of the error in bindings etc.

Reply:
Actually, I'll change the method signature to force key size. So this can be done in another PR